### PR TITLE
Improve CONTRIBUTING.md with clear installation instructions and detailed development process

### PR DESCRIPTION
**Pull request by teddyCodex**
# Description

This PR updates the CONTRIBUTING.md documentation to provide clearer instructions for setting up a development environment and contributing to the project. Key changes include:

- Added detailed steps to install and load the `usethis`, `devtools`, and `lintr` packages.
- Improved the formatting of the pull request process, making it easier to understand for new contributors.
- Updated various sections for better readability and clarity.

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality, improves documentation).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] I have added comments to my code to help provide understanding.
- [x] I have added a test which covers the code changes found within this PR.
- [x] I have deleted all non-relevant text in this pull request template.
- [x] **Reviewer assignment**: Tag a relevant team member to review and approve the changes.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -18,11 +18,34 @@ See our guide on [how to create a great issue](https://code-review.tidyverse.org
...
+    ```
+    install.packages("usethis")
+    
+    library("usethis")
+    ```
+*   Clone and fork the MolEvolvR package using:
+    ```
+    usethis::create_from_github("JRaviLab/MolEvolvR", fork = TRUE)
+    ```
+*   Install all development dependencies and then make sure the package passes R CMD check using `devtools`:
```
Comment (the-mayer) [inline .github/CONTRIBUTING.md:33]: As this package includes Bioconductor dependencies, we should probably include instructions on installing [BiocManager](https://www.bioconductor.org/install/) as well. Perhaps after this section, but before `devtools::install_dev_deps()`?

**Review by the-mayer — CHANGES_REQUESTED** [review]
Thanks for expanding the contribution guidelines. Since we are making changes, I think now might be a good time to add some additional information on getting BiocManager installed as well -- I don't think `devtools::install_dev_deps()` will manage this on it's own. 

@teddyCodex , if you could confirm and modify as appropriate, it would be appreciated!

**Comment by teddyCodex** [comment]
@the-mayer file updated.

**Side note:** 
Did not find a NEWS.md file in the repo.
Commented out the line referencing it: 
> For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in <https://style.tidyverse.org/news.html>.

Suggest restoring this line when the file exists to avoid any confusion

**Review by the-mayer — APPROVED** [review]
Good changes @teddyCodex , sending to main

**Pull request merged**